connectors/katalog/pkg/connector/connector.go

```diff
@@ -177,3 +196,63 @@ func (r *Handler) deleteAsset(c *gin.Context) {
 
 	c.JSON(http.StatusOK, &response)
 }
+
+// Enables deletion of assets to katalog.
+func (r *Handler) updateAsset(c *gin.Context) {
+	// Parse request
+	var request datacatalog.UpdateAssetRequest
+	if err := c.ShouldBindJSON(&request); err != nil {
+		errString := "Error during ShouldBindJSON in updateAsset"
+		r.log.Info().Msg(errString + err.Error())
+		r.reportError(c, http.StatusBadRequest, errString)
+		return
+	}
+	logging.LogStructure("UpdateAssetRequest received:", request, &r.log, zerolog.DebugLevel, false, false)
+
+	splittedID := strings.SplitN(string(request.AssetID), "/", 2)
+	if len(splittedID) != 2 {
+		errorMessage := fmt.Sprintf("UpdateAssetRequest has an invalid asset ID %s (must be in namespace/name format)", request.AssetID)
+		c.JSON(http.StatusBadRequest, gin.H{"error": errorMessage})
+		return
+	}
+	namespace, name := splittedID[0], splittedID[1]
+
+	asset := &v1alpha1.Asset{}
+	if err := r.client.Get(context.Background(), types.NamespacedName{Namespace: namespace, Name: name}, asset); err != nil {
+		if errors.IsNotFound(err) {
+			errString := "Error: Asset Not Found during updateAsset"
+			r.log.Info().Msg(errString + err.Error())
+			c.JSON(http.StatusNotFound, gin.H{"error": errString})
+			return
+		}
+		errString := "Error during getting asset information during update asset"
+		r.log.Info().Msg(errString + err.Error())
+		c.JSON(http.StatusInternalServerError, gin.H{"error": errString})
+		return
+	}
+
+	secretName, secretNamespace, err := vault.GetKubeSecretDetailsFromVaultPath(request.Credentials)
+	if err != nil {
+		errString := "Error obtaining a secret with credentials."
+		r.log.Info().Msg(errString + err.Error())
+		r.reportError(c, http.StatusInternalServerError, errString)
+		return
+	}
+
+	asset.Spec.Metadata = request.ResourceMetadata
+	asset.Spec.Details = request.Details
+	asset.Spec.SecretRef = v1alpha1.SecretRef{Name: secretName, Namespace: secretNamespace}
+
+	if err := r.client.Update(context.Background(), asset); err != nil {
+		r.log.Info().Msg("Error updating asset: " + err.Error())
+		c.JSON(http.StatusInternalServerError, gin.H{"Error": "Error during updating asset"})
```

Comment:
Send err.Error() instead

Reply:
Actually this change was done based on this comment by Alexey (https://github.com/fybrik/fybrik/pull/1492#discussion_r873676817).